### ISSUE TITLE
Feature: Add and remove from disables within a module

### ISSUE THREAD

**Issue by noisiver** [azerothcore/azerothcore-wotlk#9149]
### Describe your feature request or suggestion in detail

The ability to add and remove disables from within a modules would allow content to be disabled while leaving the database untouched. Removing the module wouldn't require changes to the database in order to get it back to it's original state.

### Describe a possible solution to your feature or suggestion in detail

I did add this in but due to my lacking knowledge of C++ I'm unsure it would be up to the required standards and that I'd be able to properly add request changes. It's at least a base for someone to create a PR for it or get an idea of what I'm asking for.

https://github.com/tkn963/azerothcore-wotlk/commit/8378edb22738d9f7c64ab13b05b7bff9b3413355

The idea is to add to disables on startup. My code works within WorldScript::OnStartup() but I don't know if it would from say a NPC gossip option while the world is running.

### Additional context

This is all in the name of making my progression module, which would allow adding, removing and changing content without making changes to the original tables and data.

**Comment by Nyeriah** [comment]
So basically you're proposing a function that allows adding new disables during runtime outiside the scope of the database loader, it's not a bad idea.

**Comment by tkn963** [comment]
Yes, that's correct. I didn't know how to phrase it properly, but what you just said explains it quite well.

**Comment by Nyeriah** [comment]
The only problem I see is that the point of that table was to give some flexibility. You should be able to turn things on/off during runtime without requiring a server restart, and that wouldn't be possible anymore if the disables were hardcoded. I don't think that should stop us from implementing what you propose, but it's something to be kept in mind

**Comment by tkn963** [comment]
I see what you're saying. I didn't think of that specifically, but I hope it won't pose as a problem for most users in that case.

**Comment by Err0r1** [comment]
What about a config or extra table for the module?
So you don't have to touch default AC tables

**Comment by tkn963** [comment]
That's what I'm doing for my progression module. I have custom tables for creature, creature_template, npc_vendor, gameobject, gameobject_template, disables etc. When the server starts, it will edit all data that's defined in the custom tables so nothing has to be changed within the default tables. The only other thing besides disables being added is RemoveSpellFromTrainer to match AddSpellToTrainer and a way of managing loot tables. But that's another feature request, heh.

**Comment by tkn963** [comment]
I'm not great at C++ but I'm learning as I go, if you care to see what I'm working with, feel free to check out https://github.com/tkn963/mod-progression
